plugins/microservices/src/test_msp_functions.cpp

```diff
@@ -0,0 +1,71 @@
+/// \file
+
+#include "irods/irods_ms_plugin.hpp"
+#include "irods/msParam.h"
+
+#include "msi_assertion_macros.hpp"
+
+namespace
+{
+    auto msi_impl(RuleExecInfo* _rei) -> int
+    {
+        IRODS_MSI_TEST_BEGIN("testing msp_to_ function: msp_to_string")
+        {
+            auto ret = msp_to_string(nullptr, nullptr);
+            IRODS_MSI_ASSERT(ret == SYS_NULL_INPUT);
+        }
+        {
+            char* good = nullptr;
+            auto ret = msp_to_string(nullptr, &good);
+            IRODS_MSI_ASSERT((ret == SYS_NULL_INPUT && good == nullptr));
+        }
+        {
+            msParam_t in; //NOLINT(cppcoreguidelines-init-variables)
+            in.inOutStruct = nullptr;
+            char* good = nullptr;
+            auto ret = msp_to_string(&in, &good);
+            IRODS_MSI_ASSERT((ret == SYS_NULL_INPUT && good == nullptr));
+        }
+        {
+            msParam_t in; //NOLINT(cppcoreguidelines-init-variables)
+            in.inOutStruct = malloc(sizeof(int));
+            in.type = strdup(INT_MS_T);
+            char* good = nullptr;
+            auto ret = msp_to_string(&in, &good);
+            IRODS_MSI_ASSERT((ret == USER_PARAM_TYPE_ERR && good == nullptr));
+        }
+        {
+            msParam_t in; //NOLINT(cppcoreguidelines-init-variables)
+            char* nullstr = "null";
```

Comment:
Hitting that -fpermissive flag.

ISO C++ doesn't allow string literals to be pointed to by non-const char pointers.

`char*` -> `const char*`